nearlib/account.js

```diff
@@ -1,91 +1,85 @@
+    }
 
+    /**
+     * Creates a new account with a given name and key,
+     */
+    async createAccount (newAccountId, publicKey, amount, originatorAccountId) {
+        const createAccountParams = {
+            originator: originatorAccountId,
+            new_account_id: newAccountId,
+            amount: amount,
+            public_key: publicKey,
+        };
 
-/**
- * Creates a new account with a given name and key,
- */
-exports.createAccount = async (newAccountId, publicKey, amount, originatorAccountId) => {
-    const createAccountParams = {
-        originator: originatorAccountId,
-        new_account_id: newAccountId,
-        amount: amount,
-        public_key: publicKey,
+        const transactionResponse = await this.submitTransaction("create_account", createAccountParams);
+        return transactionResponse;
     };
 
-    const transactionResponse = await submitTransaction("create_account", createAccountParams);
-    return transactionResponse;
-};
+    /**
+     * Retrieves account data by plain-text account id. 
+     */
+    async viewAccount (account_id) {
+        return await viewAccount(account_id);
+    };
 
-/**
- * Retrieves account data by plain-text account id. 
- */
-exports.viewAccount = async account_id => {
-    return await viewAccount(account_id);
-};
+    async viewAccount (account_id) {
+        const viewAccountResponse = await this.request('view_account', {
+            account_id: account_id,
+        });
+        return viewAccountResponse;
+    }
 
+    async signTransaction (transaction, sender) {
+        const stringifiedTxn = JSON.stringify(transaction);
+        const encodedKey = this.keyStore.getKey(sender).secret_key;
+        const key = bs58.decode(encodedKey);
+        const signature = [...ed25519.Sign(new Buffer(stringifiedTxn, 'utf8'), key)];
+        const response = {
+            body: transaction,
+            signature: signature
+        };
+        return response;
+    };
 
-const submitTransaction = async (method, args) => {
-    // TODO: Make sender param names consistent
-    // TODO: https://github.com/nearprotocol/nearcore/issues/287
-    const senderKeys = ['sender_account_id', 'originator_account_id', 'originator_id', 'sender', 'originator'];
-    const sender = senderKeys.map(key => args[key]).find(it => !!it)
-    const nonce = await getNonce(sender);
-    for (let i = 0; i < MAX_RETRIES; i++) {
-        const response = await request(method, Object.assign({}, args, { nonce }));
-        const transaction = response.body;
-        const signedTransaction = await signTransaction(transaction, sender);
-        const submitResponse = await request('submit_transaction', signedTransaction);
+    async submitTransaction (method, args) {
+        // TODO: Make sender param names consistent
+        // TODO: https://github.com/nearprotocol/nearcore/issues/287
+        const senderKeys = ['sender_account_id', 'originator_account_id', 'originator_id', 'sender', 'originator'];
```

Comment:
I think @ilblackdragon actually already changed all params to be `originator` here https://github.com/nearprotocol/nearcore/pull/310